core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java

```diff
@@ -1564,4 +1564,37 @@ public static boolean hasDriverAnnotations(String _cancerStudyStableIds) throws
         }
         return false;
     }
+
+    /**
+     * Returns true if there are duplicates in `mutation_event`.
+     * Compares count(*) of records in `mutation_event` against the count(distinct ...)
+     * records based on the fields specified below.
+     *
+     * If the counts do not match then SQL statement returns 0 (false), otherwise returns 1 (true).
+     * @return
+     * @throws DaoException
+     */
+    public static boolean hasDuplicateMutationEvents() throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+
+        Integer returnValue = -1; // default = -1 to indicate mismatched counts from db table
+        try {
+            con = JdbcUtil.getDbConnection(DaoMutation.class);
+            pstmt = con.prepareStatement(
+                    "SELECT "
+                    + "(SELECT COUNT(*) FROM mutation_event) = "
+                    + "(SELECT COUNT(DISTINCT ENTREZ_GENE_ID, CHR, START_POSITION, END_POSITION, TUMOR_SEQ_ALLELE, PROTEIN_CHANGE, MUTATION_TYPE) FROM mutation_event)");
+            rs = pstmt.executeQuery();
+            while (rs.next()) {
```

Comment:
I don't think a loop is needed here. The output from the query must contain only a single record I think. It would probably be safe to do:
rs.next();
returnValue = rs.getInt(1);
or if we wanted to verify that only one record was present in the resultSet we could do
if (rs.next()) {
returnValue = rs.getInt(1);
}
if (rs.next()) {
returnValue = -1; // some unexpected condition led to multiple records in the output
}

But a loop is probably not harmful, since we will always be performing a single iteration.